primitives/src/types/roles.rs

```diff
@@ -18,25 +18,42 @@ use frame_support::{dispatch::Vec, pallet_prelude::*};
 /// Role type to be used in the system.
 #[derive(Encode, Decode, Clone, Debug, PartialEq, Eq, TypeInfo)]
 pub enum RoleType {
-	Tss(TssRoleMetadata),
-	ZkSaas(ZkSaasRoleMetadata),
```

Comment:
isnt it simpler to have one singe place with all the data? why do we need both roletype and roletypemetadata?

Reply:
- It will reduce storage 
- In case of clear_role user will have to give this additional metadata which is not required.